File: src/main/java/cube8540/oauth/authentication/credentials/resource/domain/ResourceMethod.java
package cube8540.oauth.authentication.credentials.resource.domain;

public enum ResourceMethod {

    GET, POST, PUT, DELETE, ALL;

    public static ResourceMethod of(String value) {
        switch (value.toLowerCase()) {
            case "get":
                return GET;
            case "post":
                return POST;
            case "put":
                return PUT;
            case "delete":
                return DELETE;
            default:
                return ALL;
        }
    }

}

File: src/main/kotlin/cube8540/oauth/authentication/credentials/resource/domain/ResourceMethod.kt
package cube8540.oauth.authentication.credentials.resource.domain

enum class ResourceMethod {

    GET, POST, PUT, DELETE, ALL;

    companion object {
        @JvmStatic
        fun of(value: String) = when(value.toLowerCase()) {
            "get" -> GET
            "post" -> POST
            "put" -> PUT
            "delete" -> DELETE
            else -> ALL
        }
    }
}